File: src/lexer.py
# lexer.py
# Is this lexer OZER

from config import KEYWORDS

def tokenize(line):
    line = line.strip()

    if not line or line.startswith("//"):
        return [("EMPTY", "")]

    if " //" in line:
        line = line.split(" //")[0].strip()

    words = line.split()
    first_word = words[0]

    if first_word in KEYWORDS or first_word in ["}", "fdo"]:
        return [("SPECIAL", line)]

    return [("RAW_CMD", line)]

def lex(code):
    lines = code.splitlines()
    parsed_lines = []

    for line in lines:
        parsed_lines.extend(tokenize(line))

    return parsed_lines

File: src/lexer.go
package main

import (
	"strings"
)

type Token struct {
	Type  string
	Value string
}

func Tokenize(line string) []Token {
	line = strings.TrimSpace(line)

	if line == "" || strings.HasPrefix(line, "//") {
		return []Token{{"EMPTY", ""}}
	}

	if idx := strings.Index(line, " //"); idx != -1 {
		line = strings.TrimSpace(line[:idx])
	}

	return []Token{{"SPECIAL", line}}
}

func Lex(code string) []Token {
	lines := strings.Split(code, "\n")
	var parsedLines []Token

	for _, line := range lines {
		parsedLines = append(parsedLines, Tokenize(line)...)
	}

	return parsedLines
}
